### PR TITLE
Show API response in embedded kubectl view

### PR DESCRIPTION
**Pull request by n313893254**
Proposed changes
======
* Delete `cattle-node-agent` and then open kubectl view.

image

* Open kubectl view and then delete `cattle-node-agent`.

image

Types of changes
======

- New feature (non-breaking change which adds functionality)

Linked Issues
======

https://github.com/rancher/rancher/issues/15246
 

Further comments
======

### CODE REVIEW

File: lib/shared/addon/components/container-shell/component.js [hunk 1]
```diff
@@ -139,6 +140,23 @@ export default Component.extend(ThrottledResize, {
 
     set(this, 'socket', socket);
 
+    socket.onclose = (err = {}) => {
+      const term = get(this, 'term')
+
+      if (err.code === 1006) {
+        get(this, 'growl').fromError(`WebSocket connection to '${ socket.url }', failed: Error during WebSocket handshake: Unexpected response code: 404`)
```
Comment (westlywright) [inline lib/shared/addon/components/container-shell/component.js]: Localize error

Comment (vincent99) [reply lib/shared/addon/components/container-shell/component.js]: This message makes no sense, websocket code 1006 ([connection closed abnormally](https://tools.ietf.org/html/rfc6455#section-7.4.1) has nothing to do with http status 404.

**Comment by n313893254** [comment]
@westlywright Done, thanks for review both. cc @vincent99

**Comment by n313893254** [comment]
Rebased

File: lib/shared/addon/components/container-shell/component.js [hunk 2]
```diff
@@ -135,10 +137,30 @@ export default Component.extend(ThrottledResize, {
...
     const socket = new WebSocket(url, 'base64.channel.k8s.io');
 
     set(this, 'socket', socket);
 
+    socket.onclose = (err = {}) => {
+      const term = get(this, 'term')
+
+      get(this, 'growl').fromError(intl.t('containerShell.errors.close', { url: socket.url }))
+
+      if (err.code === 1006) {
```
Comment (westlywright) [inline lib/shared/addon/components/container-shell/component.js]: Is there any reason we can't make this more general? We are specifically handling error 1006 but if we check for existence of error alone and then build a generic translation message, then you should be able to pass the error code and error message to the translation so the user is informed of the exact message.

Comment (westlywright) [reply lib/shared/addon/components/container-shell/component.js]: cc: @vincent99 do you think we should make this a more general error handler? Are the URL's useful to anyone? we include it in this and the error before but the url is not escaped and looks wonky. If the url is useful we should escape it so its at least readable 

https://www.dropbox.com/s/z14dk05x0ldfqh9/Screenshot%202019-02-20%2010.03.00.png?dl=0

**Comment by vincent99** [comment]
There are lots of other error codes so I don't know why we'd special case only 1006.

I really don't think there's any meaningful information you're ever going to get out of an error from the websocket client.  They all boil down to not telling you "well it's not connected anymore".

So I would probably just make showing that it's not connected anymore more prominent instead of trying to show why.  Like graying out the background, or making the current status more obvious. You could save and show the code in that status on the off-chance it ever becomes useful to know that it's failing because code 1009.

**Comment by n313893254** [comment]
@westlywright Done, thanks for review both. cc @vincent99

**Pull request merged**